apps/music/js/db.js

```diff
@@ -39,9 +40,14 @@ function initDB() {
   });
 
   function metadataParserWrapper(file, onsuccess, onerror) {
-    LazyLoader.load('js/metadata_scripts.js', function() {
-      AudioMetadata.parse(file).then(onsuccess, onerror);
-    });
+    LazyLoader.load(
+      ['/js/metadata_scripts.js', '/js/metadata/album_art.js'],
+      function() {
+        AudioMetadata.parse(file).then(function(metadata) {
+          AlbumArt.process(file, metadata);
+        }).then(onsuccess, onerror);
```

Comment:
If it is what you intend, then maybe it is worth a comment pointing that out, since the norm seems to be that each function returns a promise that the next then waits on.